experimenter/experimenter/features/tests/test_nimbus_fml_loader.py

```diff
@@ -145,61 +157,179 @@ def test_get_fml_clients(self):
+        new_client.return_value = None
         application = "fenix"
-        path = r"\@mozilla-mobile/firefox-android/fenix/app/nimbus.fml.yaml"
-        ref = "fenix-v120.2.1"
+        path = str(self.TEST_MANIFEST_PATH)
         channel = "release"
         loader = self.create_loader(application, channel)
 
-        with mock.patch(
-            "rust_fml.FmlClient.new_with_ref",
-        ) as new_with_ref:
-            loader.create_client(path, channel, ref)
-            new_with_ref.assert_called()
+        loader._create_client(path, channel)
+        new_client.assert_called()
+        new_client.assert_called_with(path, channel)
 
-    def test_get_file_path(self):
+    def test_get_local_file_path(self):
         expected_path = Path(
-            r"\@mozilla-mobile/firefox-android/fenix/app/nimbus.fml.yaml",
+            r"/experimenter/experimenter/features/manifests/fenix/release.fml.yaml"
         )
+
+        loader = self.create_loader()
+        file_path = loader._get_local_file_path()
+        self.assertEqual(file_path, expected_path)
+
+    @patch(
+        "experimenter.features.manifests.nimbus_fml_loader.NimbusFmlLoader.MANIFEST_PATH",
+    )
+    def test_get_local_file_path_does_not_exist(self, mock_manifest_path):
+        loader = self.create_loader()
+        mock_manifest_path.return_value = Path(BASE_DIR) / "fake" / "path"
+        file_path = loader._get_local_file_path()
+        self.assertIsNone(file_path)
+
+    def test_get_remote_file_path(self):
+        expected_path = Path(r"mozilla-mobile/firefox-android/fenix/app/nimbus.fml.yaml")
         loader = self.create_loader()
-        file_path = loader.get_file_path()
+        file_path = loader._get_remote_file_path()
         self.assertEqual(file_path, expected_path)
 
-    def test_get_file_path_no_application_data(self):
+    def test_get_remote_file_path_no_application_data(self):
         loader = self.create_loader()
         loader.application_data = None
-        file_path = loader.get_file_path()
+        file_path = loader._get_remote_file_path()
         self.assertIsNone(file_path)
+
+    @patch(
+        "rust_fml.FmlClient.__init__",
+    )
+    def test_create_client(self, mock_fml_client):
+        mock_fml_client.return_value = None
+        loader = self.create_loader()
+        result = loader._create_client("some/path", "release")
+        mock_fml_client.assert_called()
+        self.assertIsInstance(result, FmlClient)
+
+    @patch(
+        "experimenter.features.manifests.nimbus_fml_loader.NimbusFmlLoader._get_fml_clients",
+    )
+    @patch(
+        "experimenter.features.manifests.nimbus_fml_loader.NimbusFmlLoader._get_inspectors",
+    )
+    @patch(
+        "experimenter.features.manifests.nimbus_fml_loader.NimbusFmlLoader._get_errors",
+    )
+    def test_get_fml_errors_fetches_client_inspector_and_error(
+        self, mock_get_clients, mock_get_inspectors, mock_get_errors
+    ):
```

Comment:
When you do multiple patch() calls, they become arguments in reverse order, e.g.

```
@patch("foo")
@patch("bar")
@patch("baz")
def test(self, mock_baz, mock_bar, mock_foo):
```

Consider one patch call:

```
@patch("bar")
def test(self, mock_bar)
```

this is equivalent to:

```
def test(self, ...):
   ...
test = patch("bar")(test)
```

When you add a second decorator:

```
@patch("foo")
@patch("bar")
def test(self, mock_bar, mock_foo)
```

this is equivalent to 

```
def test(self, ...):
    ...
test = patch("bar")(test) # replaces test(self) with test(self, mock_bar)
test = patch("foo")(test) # replaces test(self, mock_bar) with test(self, mock_bar, mock_foo)
```

See [these docs](https://docs.python.org/3/library/unittest.mock.html#quick-guide):

> Note
>
> When you nest patch decorators the mocks are passed in to the decorated function in the same order they applied (the normal Python order that decorators are applied). This means from the bottom up, so in the example above the mock for module.ClassName1 is passed in first.
>
> With [patch()](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) it matters that you patch objects in the namespace where they are looked up. This is normally straightforward, but for a quick guide read [where to patch](https://docs.python.org/3/library/unittest.mock.html#where-to-patch).